### PR TITLE
Pull Request for TSE Review of Bubble Sort Solution

### PR DESCRIPTION
**Pull request by Calvinoea**
Hello. We (@hillarioh and I) are making a pull request for review of our solution for the Bubble Sort project/task as part of the Microverse/Odin Curriculum. 

The solution has been made using Ruby. Stickler and rubucop were used to check for errors in the code. The code has passed tests by stickler (on Github) and rubocop. 

The solution includes a bubble sort and bubble sort by method.

Please find below a link to the demo of the solution online: 

[Demo](https://repl.it/@CalvinEbun_Amu/Bubble-Sort)

Thank you for your time and review.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,52 @@
...
+Type ruby bubblesort.rb through the terminal to run the code or use the debug button in your text editor.
+
+## 🤝 Contributing
+You are welcome to make contributions to the repository. Contributions may be made through issues comments and feature requests.
+
+
+## 👤 Author
+
+# Hillary Okerio
+- GitHub [here](https://github.com/hillarioh/)
```
Comment (mosesogwo) [inline README.md]: It'd be better to use your GitHub handle instead of "here"

File: README.md [hunk 2]
```diff
@@ -1 +1,52 @@
...
+- Twitter: [@hillaokri](https://twitter.com/hillaokri)
+
+# Calvin
+- GitHub [here](https://github.com/calvinoea/)
```
Comment (mosesogwo) [inline README.md]: It'd be better to use your GitHub handle instead of "here"

File: bubblesort.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+def bubble_sort(array)
+  laps = 0
+
+  while laps < array.size - 1
+
+    (0...array.size - 1).each do |item|
+      array[item], array[item + 1] = array[item + 1], array[item] if array[item] > array[item + 1]
+    end
+    print array
+    puts ''
```
Comment (mosesogwo) [inline bubblesort.rb]: Remove the print and puts. The return of the sorted array is sufficient.

**Review by mosesogwo — CHANGES_REQUESTED** [review]
**HI @Calvinoea, @hillarioh, good work :+1:**

Enable stickler for this repository and ensure that all tests pass and please see my comments on your files.

_**Kindly make those changes and submit another code review request.**_

**Any questions? You can reach me on Slack (Moses Ogwo)**

**Cheers:stars:**

### Status: Changes Requested.

**Review by majovanilla — CHANGES_REQUESTED** [review]
Hi team! :wave: 

## You did a *great* job on this project! 💜 

However, I can't see the bubble_sort_by method in your project!!

You need to implement this method, too before approving your project.

Please work on this requirement and request a new code review! :smile: 


If you have any questions, feel free to send me a message through Slack (@majo)
I would be happy to help! 👩‍💻 


Happy coding 🤓

--Maria Reyes

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Great Job
You have made the requested changes by the previous reviewers but we still need to fix some failing edge cases before we can merge this PR.
- [x] enable stickler in this repo and make sure all tests pass
- [ ] add a `bubble_sort` method. We need two working methods in this repo.
- [x] check my comments for more suggestions

### Martin Afani

File: bubblesort.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  while laps < array.size - 1
+
+    (0...array.size - 1).each do |item|
+      array[item], array[item + 1] = array[item + 1], array[item] if array[item] > array[item + 1]
+    end
+
+    laps += 1
+  end
+  array
+end
```
Comment (whiz25) [inline bubblesort.rb]: - [x] bubble_sort_by method is implemented and working (Screenshot from Odin)
- [x] sorts an array of numbers in increasing order
```ruby
unsorted = (1..10).to_a.reverse!
bubble_sort(unsorted) == unsorted.sort # should return true
```
- [x] sorts an array by accepting a block
```ruby
unsorted = %w[hi hello hey]
bubble_sort_by(unsorted) { |left, right| left.length - right.length } == %w[hi hey hello] # should return true
```
- [x] raises an error if block is not given
```ruby
unsorted = %w[hi hello hey]
bubble_sort_by(unsorted) == raise_error # should return true
```

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great progress @Calvinoea :clap: :clap: However, some of the changes requested by the previous reviewer were not implemented. Check :arrow_down: 

* [ ] bubble_sort_by method is implemented and working ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_bubble_sort/sortBy.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-1))
- You need to come up with your own version of **bubble_sort_by** that accept blocks and raises an error when no block is given. Read the description in the link above, It should help you understand what are the requirements for this method.

### That's it :relieved: When finished, kindly resubmit another code review request :pray: 

Happy coding :v: 
**_Ahmed Mahfoudh_**

**Review by mosesogwo — APPROVED** [review]
**Hi @Calvinoea, Congratulations! Your project is approved:trophy:**

You can now merge to master

**Cheers:stars:**

### Status: Approved :heavy_check_mark:
